eclair-core/src/test/scala/fr/acinq/eclair/integration/IntegrationSpec.scala

```diff
@@ -825,6 +827,62 @@ class IntegrationSpec extends TestKit(ActorSystem("test")) with BitcoindService
+    val sender = TestProbe()
+    sender.send(nodes("F6").register, 'channelsTo)
+    // retrieve the channelId of C <--> F6
+    val Some(channelId) = sender.expectMsgType[Map[ByteVector32, PublicKey]].find(_._2 == nodes("C").nodeParams.nodeId).map(_._1)
+
+    sender.send(nodes("F6").register, Forward(channelId, CMD_GETSTATEDATA))
+    val initialStateDataF6 = sender.expectMsgType[DATA_NORMAL]
+    val initialCommitmentIndex = initialStateDataF6.commitments.localCommit.index
+
+    // the 'to remote' address is a simple P2WPKH spending to the remote payment basepoint
+    val toRemoteAddress = Script.pay2wpkh(initialStateDataF6.commitments.remoteParams.paymentBasepoint)
+
+    // toRemote output of C as seen by F6
+    val Some(toRemoteOutC) = initialStateDataF6.commitments.localCommit.publishableTxs.commitTx.tx.txOut.find(_.publicKeyScript == Script.write(toRemoteAddress))
+
+    // let's make a payment to advance the commit index
+    val amountMsat = 4200000.msat
+    sender.send(nodes("F6").paymentHandler, ReceivePayment(Some(amountMsat), "1 coffee"))
+    val pr = sender.expectMsgType[PaymentRequest]
+
+    // then we make the actual payment
+    sender.send(nodes("C").paymentInitiator, SendPaymentRequest(amountMsat, pr.paymentHash, nodes("F6").nodeParams.nodeId, routeParams = integrationTestRouteParams, maxAttempts = 1))
+    val paymentId = sender.expectMsgType[UUID](5 seconds)
+    val ps = sender.expectMsgType[PaymentSent](5 seconds)
+    assert(ps.id == paymentId)
+
+    sender.send(nodes("F6").register, Forward(channelId, CMD_GETSTATEDATA))
+    val stateDataF6 = sender.expectMsgType[DATA_NORMAL]
+    val commitmentIndex = stateDataF6.commitments.localCommit.index
+    val commitTx = stateDataF6.commitments.localCommit.publishableTxs.commitTx.tx
+    val Some(toRemoteOutCNew) = commitTx.txOut.find(_.publicKeyScript == Script.write(toRemoteAddress))
+
+    // there is a new commitment index in the channel state
+    assert(commitmentIndex == initialCommitmentIndex + 1)
+
+    // script pubkeys of toRemote output remained the same across commitments
+    assert(toRemoteOutC.publicKeyScript == toRemoteOutCNew.publicKeyScript)
+
+    // now let's force close the channel and check the toRemote is what we had at the beginning
+    sender.send(nodes("F6").register, Forward(channelId, CMD_FORCECLOSE))
+    sender.expectMsg(ChannelCommandResponse.Ok)
+    // we then wait for C to detect the unilateral close and go to CLOSING state
+    awaitCond({
+      sender.send(nodes("C").register, Forward(channelId, CMD_GETSTATE))
+      sender.expectMsgType[State] == CLOSING
+    }, max = 20 seconds, interval = 1 second)
+
+    sender.send(bitcoincli, BitcoinReq("getrawtransaction", commitTx.txid.toHex))
+    val JString(rawTx) = sender.expectMsgType[JValue](10 seconds)
+
+    // the unilateral close contains the static toRemote output
+    assert(Transaction.read(rawTx).txOut.exists(_.publicKeyScript == toRemoteOutC.publicKeyScript))
```

Comment:
Can we also test that the channel correctly goes to the `CLOSED` state?

Reply:
Yes it makes sense especially because with static remote key we don't make claim transaction so it's worth to test that we move to `CLOSED`  , done in 4abf88d